spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/security/IapAuthenticationAutoConfiguration.java

```diff
@@ -54,18 +71,73 @@
 @ConditionalOnProperty("spring.cloud.gcp.security.iap.enabled")
 @ConditionalOnClass({NimbusJwtDecoderJwkSupport.class})
 @AutoConfigureBefore(OAuth2ResourceServerAutoConfiguration.class)
+@AutoConfigureAfter(GcpContextAutoConfiguration.class)
 @EnableConfigurationProperties(IapAuthenticationProperties.class)
 public class IapAuthenticationAutoConfiguration {
 
+	/**
+	 * Provides default implementation for determining GCP environment.
+	 * Can be overridden to avoid interacting with real environment.
+	 */
+	@Bean
+	@ConditionalOnMissingBean
+	public GcpEnvironmentProvider gcpEnvironmentProvider() {
```

Comment:
I think this one must be `static` to avoid some early `BeanFactory` access when we evaluate condition.